packages/compass-aggregations/src/modules/pipeline-builder/pipeline-mode.ts

```diff
@@ -42,19 +42,17 @@ export const changePipelineMode = (
   newMode: PipelineMode
 ): PipelineBuilderThunkAction<void, PipelineModeToggledAction> => {
   return (dispatch, getState, { pipelineBuilder }) => {
+    if (newMode === getState().pipelineBuilder.pipelineMode) {
```

Comment:
Drive-by, this doesn't really cause any issues in the app, but still lead to telemetry events being fired more than needed